data/spells/lib/spells.lua

```diff
@@ -259,6 +259,10 @@ RANGE = {
 }
 
 function Creature:addDamageCondition(target, conditionType, listType, damage, time, rounds)
+	if self:isImmune(conditionType) then
```

Comment:
Oh, I see now, you are right, it should be the target lol. By the way I also update the script to use ``math.pow`` instead of ``^``.